### ISSUE TITLE
plugins/LadspaEffect/swh/ladspa-util.c is unused and has build problems with new gcc

### ISSUE THREAD

**Issue by plater** [LMMS/lmms#5434]
I maintain lmms for openSUSE and found an irritating problem with ladspa-util.c, regardless of flags used to link it, it ends up as a static library. This is gcc-9.2.1+git1192 and binutils-2.34.
During my 1 days head scratch I thought that sed'ing out the -Wl,--as-needed flag fixed the problem but it still happened with a cloud build service build, it causes a warning because of the .so extension.
At the end of the day I searched for usages and found none, I now delete it pre build. I see that you use Ladspa's git for master and Ladspa-util.c doesn't exist in Ladspa only Ladspa-util.h, maybe it should be removed from the 1.2 branch. The decision is yours.

**Comment by JohannesLorenz** [comment]
@plater Thanks for your efforts and sorry for our later answer. I use gcc-9.3.0 and binutils-2.34 (same version as yours). On my machine, it all compiles fine.

What exactly do you delete pre build? The `plugins/LadspaEffect/swh/ladspa-util.h` on branch stable-1.2? This file seems to be required.

All files I found:
```
$ find {master,stable-1.2}/plugins/ -iname "*ladspa-util*"
master/plugins/LadspaEffect/swh/ladspa/ladspa-util.h
stable-1.2/plugins/LadspaEffect/swh/ladspa-util.c
stable-1.2/plugins/LadspaEffect/swh/ladspa-util.h
stable-1.2/plugins/LadspaEffect/swh/ladspa/ladspa-util.h # somehow not under version control
```

**Comment by plater** [comment]
ladspa-util.h is part of the current ladspa headers but I can't find ladspa-util.c anywhere and it fails to build under gcc9. I suspect that it comes from an older release of ladspa.
I remove stable-1.2/plugins/LadspaEffect/swh/ladspa-util.c pre build.
Another question, could you possibly allow an external ladspa for the build to replace the third party ladspa?

**Comment by JohannesLorenz** [comment]
> ladspa-util.h is part of the current ladspa headers but **I can't find ladspa-util.c** anywhere and it fails to build under gcc9. I suspect that it comes from an older release of ladspa.
> **I remove stable-1.2/plugins/LadspaEffect/swh/ladspa-util.c** pre build.

I was just thinking "How can you remove it if you cannot find it?" :smiley: But I think you mean you can't find it in swh/ladspa, so you remove it in LMMS, too? I just tried that and LMMS still compiles. The file seems useless. I'll remove the file in PR #5451 - thanks for the report.

> Another question, could you possibly allow an external ladspa for the build to replace the third party ladspa?

On stable-1.2, no chance. On master, we'll do this for the next release. For some LADSPA, like CALF and CAPS, using their exact master won't be possible (they introduced backwards incompatibility, so we wrote our own versions), we had to fork them. So you can't always use the system LADSPA library. We'll know more when the next LMMS master release will come.

**Comment by JohannesLorenz** [comment]
Fixed in #5451
